### PR TITLE
Added textDocument/foldingRange

### PR DESCRIPTION
**Pull request by SasLuca**
A simple initial implementation of `textDocument/foldingRange` that enables folding matching multi-line curly brace ranges.

### CODE REVIEW

**Review by SuperAuguste — CHANGES_REQUESTED** [review]

File: src/Server.zig [hunk 1]
```diff
@@ -2339,6 +2339,89 @@ fn codeActionHandler(server: *Server, writer: anytype, id: types.RequestId, req:
     });
 }
 
+fn foldingRangeHandler(server: *Server, writer: anytype, id: types.RequestId, req: requests.FoldingRange) !void
+{
+    const Tag = std.zig.Token.Tag;
+    const allocator = server.arena.allocator();
+    const handle = server.document_store.getHandle(req.params.textDocument.uri) orelse {
+        log.warn("Trying to get inlay hint of non existent document {s}", .{req.params.textDocument.uri});
```
[the remark this answers is not in the document]
Comment (SasLuca) [reply src/Server.zig]: Yeah, silly copy paste issue. Fixed it!

**Comment by leecannon** [comment]
I actually get less folding regions with this PR than without it. I assume this is to do with having https://marketplace.visualstudio.com/items?itemName=tiehuis.zig as well as zls-vscode.

## Without
old

## With
new

**Comment by SasLuca** [comment]
I get no folding regions at all without this change. I use both the zig and zls-vscode plugins. Do you use just one?

Seems like the difference between the 2 images is that multiline parens pairs are also given a fold option in the first one, I could add this easily.

**Comment by leecannon** [comment]
I use both the zig and zls extensions, as far as I can remember i've always had folding regions no idea why.

**Comment by InKryption** [comment]
@leecannon Probably the current indentation you're seeing is because you have `"editor.foldingStrategy": "indentation"`, whereas @SasLuca probably has `"editor.foldingStrategy": "auto"`. Overall I think this is going to be better, because this can avoid providing folding regions that don't make sense.

**Comment by SasLuca** [comment]
Yeah, I think the main issue is just what regions we want to provide folding for. Right now I do it for scopes and for comments like `//#region`. Will probably add support for multi-line comments and maybe for paren pairs that span multiple lines.

**Review by SuperAuguste — APPROVED** [review]
LGTM! :)

**Pull request merged**